resistor-colors: add new exercise (#1466)
exercism/problem-specifications: exercises/resistor-significants/description.md

> If you would build anything with a Raspberry Pi or an Arduino, one of the first components you need are resistors. In this exercise, it doesn't matter what a resistor is or does, the only thing that matters is that they come with different values. And that they are tiny. So tiny, that it's hard to read such a value if it were printed on the thing. To solve that, fabricators use color codes that reveal their value. Which is great, but it makes it hard to see the actual value.

Reviewer: I have some thoughts on rewording this but I don't have the time at the moment.  This evening I'll revisit this and update this comment...

Reviewer: I wasn't paying very careful attention to what comes after what I just wrote.  Some of it may be redundant.

Author: @F3PiX Could you look at the suggested changes?

Reviewer 2: Thanks @rpottsoh ! 
It's not so easy to distinguish the separate changes in texts with the Suggested Changes feature.  What I see, there's some corrections of mistakes (thanks!). There's also some level of detail added that I'd rather keep out. Let me see how I can work the changes in.

> If you would build anything with a Raspberry Pi or an Arduino, one of the components you will likely use are resistors. In this exercise, it doesn't matter what a resistor is or how it works, the only thing that matters is that a resistor has a 'value'. And that they are small. So small, that it's hard to read such a value if it were to be actually printed on it. To solve that, fabricators use color codes that denotes the resistor's value. Which is great, but it makes it hard to see the actual value.

Reviewer 3: I've tweaked this below. The main changes:
-  I've restructured it to make it flow a little more clearly, and to fill in a few blanks/unknowns that I had as I read it (I found myself googling about resistors to clarify things). 
- I've removed the "or an Arduino" because I don't think we need both and Raspberry Pis are more well know and - I changed the example to brown and green, because the brown/black `10` thing made me think of binary and confused me :)
- I changed "fabricators" to "manufacturers" as I think it's a more common word.
- I specified "resistance value" because I found myself asking "what/why does this have a value" whereas I could immediately visualise "resistance value" as a tangible thing.

```
If you want to build something using a Raspberry Pi, you'll probably use *resistors*. For this exercise, you need to know two things about them:
1. Each resistor has a resistance value. 
2. Resistors are small - so small in fact that if you printed the resistance value on them, it would be impossible to read.  

To get around this problem, manufacturers print color-coded bands onto the resistors to denote their resistance values. Each band acts as a digit of a number. For example, if they printed a brown strip (value 1) followed by a green strip (value 5), the resistance value of the resistor would be 15.

In this exercise, you are going to create a helpful program so that you don't have to remember the values of the bands. The program will take two colors as input, and output the correct resistance value.

The band colors are encoded as follows:

- Black: 0
- Brown: 1
- Red: 2
- Orange: 3
- Yellow: 4
- Green: 5
- Blue: 6
- Violet: 7
- Grey: 8
- White: 9
```

Would appreciate everyone's thoughts - especially @kytrinyx and @f3pix. I've not done it as a "suggestion" so I'll leave you to manually put it in the PR if you like ti.

Reviewer 4: This looks good to me! (I know nothing about resistors)

Reviewer: These suggestions overall seem fine to me.

> I changed "fabricators" to "manufacturers" as I think it's a more common word.

How about `makers` instead?

Personally if I didn't know anything about resistors having some basic facts included is helpful, adds some color to the exercise (I know, but pun wasn't intended) , makes the "thing" less nebulous even if the minor information isn't directly relateable it helps me to focus what my mind should be considering while solving the problem; as long as the problem description goes on to indicate which information is essential for completing the exercise.

> Resistors have color coded bands, where each color maps to a number. The first 2 bands of a resistor have a simple encoding scheme: each color maps to a single number.

Reviewer 2: ```suggestion
Resistors have color coded bands that encode their value in Ohm. The bands have a simple encoding scheme: each color maps to a single number. The first 2 bands represent the significant figures of a resistor's value in Ohm. 
[Significant figures](https://en.wikipedia.org/wiki/Significant_figures) or significant digits is a concept used in measurements to indicate the numbers without trailing zero's.   
```

These colors are encoded as follows:

- Black: 0
- Brown: 1
- Red: 2
- Orange: 3
- Yellow: 4
- Green: 5
- Blue: 6
- Violet: 7
- Grey: 8
- White: 9

Combining the numbers of the first two bands gives one its significant value. As an example, if the first band is blue (6) and the second band is grey (8), the significant value is 68. 

> 

Reviewer 2: ```suggestion
Combining the numbers of the first two bands gives one the significant figures of the value in Ohm.
```
And in the example: the value is `68`

Reviewer 2: Or: the value to be returned is `68` . (To avoid the problem of what we should call this part. I'm afraid significant value may be misleading or confusing for its meaning in statistics. )

Resistors have color coded bands that encode their value in Ohms. The bands have a simple encoding scheme: each color maps to a single number. The first 2 bands represent the significant figures of a resistor's value in Ohms. 
[Significant figures](https://en.wikipedia.org/wiki/Significant_figures) or significant digits is a concept used in measurements to indicate the numbers without trailing zero's.

The band colors are encoded as follows:

- Black: 0
- Brown: 1
- Red: 2
- Orange: 3
- Yellow: 4
- Green: 5
- Blue: 6
- Violet: 7
- Grey: 8
- White: 9

Combining the numbers of the first two bands gives one the significant figures of its value in Ohms. As an example, if the first band is blue (6) and the second band is grey (8), the combined significant figures' value is the number 68. 

> 

Reviewer 2: ```suggestion
If you would build anything with a Raspberry Pi or an Arduino, one of the first components you need are resistors. In this exercise, it doesn't matter what a resistor is or does, the only thing that matters here is that they come with different values. And that they are tiny. So tiny that it's hard to read such a value if it were printed on the thing. To solve that, fabricators use color codes that reveals their value. Which is great, but it makes it hard to see the actual value.
You are going to solve this with a program that you can feed two colors, and it will translate the color to a number. For instance "brown" and "black" are code for `1` and `0`, and together they represent the value `10`.

The band colors are encoded as follows:

Black: 0
Brown: 1
Red: 2
Orange: 3
Yellow: 4
Green: 5
Blue: 6
Violet: 7
Grey: 8
White: 9

```
So, I removed all the tech talk ;-) including the part about the Ohms value. I feel it's important to not suggest that we're decoding Ohms value, because that would confuse people who do know about resistors.

Reviewer 4: I agree! Good choice.

> More information on the color encoding of resistors can be found in the [Electronic color code Wikipedia article](https://en.wikipedia.org/wiki/Electronic_color_code)

Reviewer 2: And shall we take this out too? It's not helpful in solving the exercise.

Reviewer 4: Yeah, let's take it out. People can google it if they want to!

If you want to build something using a Raspberry Pi, you'll probably use _resistors_. For this exercise, you need to know two things about them:

* Each resistor has a resistance value.
* Resistors are small - so small in fact that if you printed the resistance value on them, it would be hard to read.

> To get around this problem, manufacturers print color-coded bands onto the resistors to denote their resistance values. Each band acts as a digit of a number. For example, if they printed a brown strip (value 1) followed by a green strip (value 5), it would translate to the number 15.

Reviewer: Are they bands or strips of color?

Reviewer: I like what I am reading.  :+1:

- Grey: 8
- White: 9

> Combining the numbers of the first two bands gives one the significant figures of its value in Ohm. As an example, if the first band is blue (6) and the second band is grey (8), the combined significant figures in Ohm equal the number 68. 

Reviewer 2: 👍 Better, thanks. I think 'in Ohm' can go too, to make it even more clear.

Reviewer: Ohm should be Ohms.

Reviewer 2: I googled that ;-) But I trust you way more than google.

Author: @rpottsoh should we use Ohms everywhere?

Reviewer: Think of Ohms as cookies.  Conversationally you wouldn't say you have 5 cookie, but rather 5 cookies.  How many cookies in a dozen?....  Having singular Ohm in your sentence(s) may not be technically wrong, but it "sounds" odd when spoken out loud.  How many Ohms is the resistor?  It is 5 Ohms or it is 1 Ohm or it is 1/2 [an] Ohm.  Use plural when the resistor is more than 0 and <= 1.  Then there is a twist, when asking a question....  Do you have a 5 Ohm resistor I can borrow?  Ohm is singular.

Reviewer 5: > Then there is a twist, when asking a question.... Do you have a 5 Ohm resistor I can borrow? Ohm is singular.

:+1: Like a 5-cookie jar.

Reviewer: Ha, thanks for keeping the analogy going.  :smile:

> 

Reviewer 2: 👍 Better, thanks. I think 'in Ohm' can go too, to make it even more clear.
